dts/arm/infineon/xmc4700_F144x2048.dtsi

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (c) 2023 Schlumberger
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <mem.h>
+#include <infineon/xmc4xxx.dtsi>
+
+/ {
+	psram1: memory@1ffe8000 {
+		compatible = "mmio-sram";
+		reg = <0x1ffe8000 DT_SIZE_K(96)>;
+	};
+
+	dsram1: memory@20000000 {
+		compatible = "mmio-sram";
+		reg = <0x20000000 DT_SIZE_K(128)>;
+	};
+
+	dsram2: memory@20020000 {
+		compatible = "mmio-sram";
+		reg = <0x20020000 DT_SIZE_K(128)>;
+	};
+};
+
+&flash0 {
+	reg = <0xc000000 DT_SIZE_M(2)>;
+	pages_layout: pages_layout {
+		pages_layout_16k: pages_layout_16k {
+			pages-count = <8>;
+			pages-size = <DT_SIZE_K(16)>;
+		};
+		pages_layout_128k: pages_layout_128k {
+			pages-count = <1>;
+			pages-size = <DT_SIZE_K(128)>;
+		};
+		pages_layout_256k: pages_layout_256k {
+			pages-count = <7>;
+			pages-size = <DT_SIZE_K(256)>;
+		};
+	};
+};
+
+&gpio0 {
+	ngpios = <16>;
+};
+
+&gpio1 {
+	ngpios = <16>;
+};
+
+&gpio2 {
+	ngpios = <16>;
+};
+
+&gpio14 {
+	ngpios = <14>;
+};
+
+&gpio15 {
+	ngpios = <12>;
```

Comment:
@npal-cy, good catch. I think your proposed solution is the cleanest way to do it. I'll update the PR in the next couple of days.

Reply:
@npal-cy dts/bindings/gpio-controller.yaml covers how to handle these cases (see gpio-reserved-ranges). I've updated the PR based on the .yaml file.

Reply:
@talih0 , you still need to update gpio driver to get correct .port_pin_mask. Currently you are using `.common = {.port_pin_mask = GPIO_PORT_PIN_MASK_FROM_DT_INST(index)}};' and GPIO_PORT_PIN_MASK_FROM_DT_INST does't  handle this.

please look at drivers\gpio\gpio_nxp_s32.c, they also had same case.

Reply:
@npal-cy, I think the proper fix is to fix the GPIO_PORT_PIN_MASK_FROM_DT_INST() macro. Otherwise, each driver will have their own implementation:
- gpio_nxp_s32.c is not great because it can only handle one reserved region. 
- gpio_mcux_igpio.c has a runtime solution, but I think it's better to handle this at compile time.

But currently, I don't think it's possible to merge multiple gpio-reserved-ranges into a common reserved_mask at compile time. Initially I thought we could use LISTIFY(LEN, ....), but unfortunately this doesn't work because LEN needs to be literal constant and we need to iterate over half of the array gpio-reserved-ranges array.

Overall, I think fixing this issue should be a separate PR that will be used cleanly by other gpio drivers.

Reply:
could you please file ticket for GPIO_PORT_PIN_MASK_FROM_DT_INST update?